app/controllers/application_controller.rb

```diff
@@ -20,9 +20,9 @@ class ApplicationController < ActionController::Base
   end
 
   before_action :set_timezone
-
   before_action :enable_analytics
   before_action :check_browser
+  before_action :set_cache_headers
```

Comment:
Shouldn't we apply this only in certain situations where sensitive data might be preserved locally?

Reply:
I agree with @akostadinov. Disabling the cache everywhere has side effects:

- The overall performance will be affected since all requests must be always sent to the server.
- Costs will increase for providers, because of the increase of requests received.

I think we should keep the cache enabled by default and disable it only for a white list of screens that include auth data, like tokens. I would do it this way:

1. Add a new method to `ApplicationController`: `disable_client_cache?` which returns `false`
2. Rename `set_cache_hesders` to `disable_client_cache`, it's more clear
3. Call `disable_client_cache` only when `disable_client_cache?` is `true`. Like this:
```
before_action :disable_client_cache, if: :disable_client_cache?
```
4. Define the method `disable_client_cache?` to return `true` in those particular controllers that include auth data.

The only problem is to find that list of controllers. The next come to my mind:

**Admin portal:**
- `/p/admin/api_docs`
- `/p/admin/account`
- `/p/admin/user/access_tokens`
- `/p/admin/account/authentication_providers` <- Maybe? not sure about this one
- `/p/admin/webhooks/` <- Could include tokens in the URL as paremeters
- `/apiconfig/services/<ID>/settings`
- `/apiconfig/services/<ID>/integration`
- `/apiconfig/services/<ID>/api_docs/<ID>/preview`
- `/p/admin/applications/<ID>`
- `/site/dns`
- `/p/admin/authentication_providers/<ID>/edit`
- `/p/admin/authentication_providers/new`

**Developer portal:**
- `/`
- `/admin/applications/<ID>`
- `/docs`

The developer portal is tricky because URLs can change, and we can't control what users show in their templates anyway, so the whole developer portal should have cache disabled probably.

Since this should be merged soon, I think it's OK to merge this now and work on the fine-grain filtering of controllers later.

Reply:
> /p/admin/account/authentication_providers <- Maybe? not sure about this one

Yeah, while `#index` seems safe, but `#show` has the provider's client ID and secret.
So, I think we can add it.

> /p/admin/webhooks/ <- Could include tokens in the URL as paremeters

Agreed


Also,

`/p/admin/api_docs/account_data` - it returns user's credentials

`/finance/settings` - shows the payment gateway settings